app/logging/Logging.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package logging
+
+import play.api.Logger
+
+trait Logging {
+
+  protected val logger: Logger = Logger(s"application.${this.getClass.getCanonicalName}")
```

Comment:
Being a val how does this change for each class? Would you mind quickly going over this with me.

Reply:
This is on a trait that various classes extend.  The val gets initialised on the class itself, not on the trait - so `this.getClass` is actually being called on e.g. `ReferenceDataListController` etc.